src/stdpar/CondFormats/SiPixelGainCalibrationForHLTGPU.cc

```diff
@@ -10,28 +10,17 @@
 #include "CUDACore/StreamCache.h"
 
 SiPixelGainCalibrationForHLTGPU::SiPixelGainCalibrationForHLTGPU(SiPixelGainForHLTonGPU const& gain,
-                                                                 std::vector<char> const& gainData) {
-  cudaCheck(cudaMallocManaged(&gainForHLT_, sizeof(SiPixelGainForHLTonGPU)));
-  *gainForHLT_ = gain;
-  cudaCheck(cudaMallocManaged(&gainData_, gainData.size()));
-  gainForHLT_->v_pedestals = gainData_;
-
-  std::memcpy(gainData_, gainData.data(), gainData.size());
-  for (int device = 0, ndev = cms::cuda::deviceCount(); device < ndev; ++device) {
-#ifndef CUDAUVM_DISABLE_ADVISE
-    cudaCheck(cudaMemAdvise(gainForHLT_, sizeof(SiPixelGainForHLTonGPU), cudaMemAdviseSetReadMostly, device));
-    cudaCheck(cudaMemAdvise(gainData_, gainData.size(), cudaMemAdviseSetReadMostly, device));
+                                                                 std::vector<DecodingStructure> const& gainData)
+    : gainForHLT_{std::make_unique<SiPixelGainForHLTonGPU>(gain)},
+//make_shared for array
+#if __GNUC__ >= 12
+      gainData_{std::make_shared<DecodingStructure[]>(gainData.size())}
+#else
+      gainData_{new DecodingStructure[gainData.size()]}
```

Comment:
The `gainData` could probably be also `std::vector<DecodingStructure>`, right? That would help to avoid the `#if`. (otherwise, why `shared_ptr` instead of `unique_ptr?)